src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
```

Comment:
1. you can create value with mothers array and then filter people. 
2. you can add withChildren validation in .filter() method:
Example: 
```suggestion
  const mothers = people.map(person => person.mother);
  const women = people.filter(person => (
    withChildren
      ? mothers.includes(person.name) && person.sex === 'f'
      : person.sex === 'f';
  ));
```